File: backend/backend.php
<?php
// STEP 1
// Load dependency
require __DIR__ . '/vendor/autoload.php';

// Load json file
$batch = json_decode(file_get_contents('../data/articles.json'), true);
var_dump($batch);

// Initialize API Client
$client = new \AlgoliaSearch\Client('YOUR_APP_ID', 'YOUR_ADMIN_API_KEY');

// Initialize index
$index = $client->initIndex('smashing');

// Push data
$index->addObjects($batch);

// STEP 2
// Set settings
$index->setSettings(array(
		"searchableAttributes" => array("title", "description", "tags", "author"),
		"customRanking" => array("desc(commentCount)")));

// Set settings and get settings
$res = $index->setSettings(array("attributesForFaceting" => array("tags.name")));
$index->waitTask($res['taskID']); // as operations are asynchronous, we explicitly wait for task completion
$settings = $index->getSettings();
var_dump($settings['attributesForFaceting']);
?>

File: backend/backend.py
#!/usr/bin/env python
import json
from algoliasearch import algoliasearch

# Step 1

# Get data
with open("../data/articles.json") as f:
    articles = json.load(f)
    # print(json.dumps(articles, indent=4))

# Init client
client = algoliasearch.Client("YOUR_APP_ID", "YOUR_ADMIN_API_KEY")

# Create index
index = client.init_index("smashing")

# Push data
index.add_objects(articles)


# Step 2

# Set settings: searchable attributes and ranking (async)
index.set_settings({
    "searchableAttributes": ["title", "description", "tags", "author"],
    "customRanking": ["desc(commentCount)"]
})

# Set settings: faceting on tag names (sync)
res = index.set_settings({
    "attributesForFaceting": ["tags.name"]
})
index.wait_task(res['taskID'])
print("Attributes for faceting: %s." % index.get_settings()['attributesForFaceting'])


